spectrochempy/core/readers/importer.py

```diff
@@ -492,13 +495,15 @@ def _read_dir(*args, **kwargs):
     directory = get_directory_name(directory)
     files = get_filenames(directory, **kwargs)
     datasets = []
-    for key in files.keys():
+    valid_extensions = list(zip(*FILETYPES))[0] + list(zip(*ALIAS))[0]
+    for key in [key for key in files.keys() if key[1:] in valid_extensions]:
         if key:
             importer = Importer()
             nd = importer(files[key], **kwargs)
-            if not isinstance(nd, list):
-                nd = [nd]
-            datasets.extend(nd)
+            if nd is not None:
```

Comment:
Are you sure you need this. For me it works well without this?  Can you provide an example when it doesn't work?

Reply:
Yes it was not necessary :-)